Venmo/src/main/java/com/braintreepayments/api/VenmoLauncher.java

```diff
@@ -0,0 +1,63 @@
+package com.braintreepayments.api;
+
+
+import androidx.activity.result.ActivityResultLauncher;
+import androidx.activity.result.ActivityResultRegistry;
+import androidx.annotation.NonNull;
+import androidx.annotation.VisibleForTesting;
+import androidx.fragment.app.Fragment;
+import androidx.fragment.app.FragmentActivity;
+import androidx.lifecycle.LifecycleOwner;
+
+/**
+ * Responsible for launching the Venmo app to authenticate users
+ */
+public class VenmoLauncher {
+
+    @VisibleForTesting
+    ActivityResultLauncher<VenmoAuthChallenge> activityLauncher;
+    private static final String VENMO_SECURE_RESULT = "com.braintreepayments.api.Venmo.RESULT";
+
+    /**
+     * Used to launch the Venmo authentication flow to tokenize a Venmo account. This class must
+     * be instantiated in the OnCreate method of your Fragment.
+     *
+     * @param fragment an Android Fragment from which you will launch the Venmo app
+     * @param callback a {@link VenmoAuthChallengeResultCallback} to receive the result of the
+     *                 Venmo app switch authentication flow
+     */
+    public VenmoLauncher(@NonNull Fragment fragment,
+                         @NonNull VenmoAuthChallengeResultCallback callback) {
+        this(fragment.getActivity().getActivityResultRegistry(), fragment.getViewLifecycleOwner(),
```

Comment:
Question - is `fragment.getViewLifecycleOwner()` something that would be acceptable for a merchant to do? Just curious if we need to offer 2 VenmoLauncher inits, or if doing so prevents issues.

Reply:
That's really helpful, thanks Sarah!